sdk/core/Azure.Core/src/Internal/AzureBaseBuffersExtensions.cs

```diff
@@ -87,5 +90,50 @@ public static async Task WriteAsync(this Stream stream, ReadOnlySequence<byte> b
                     ArrayPool<byte>.Shared.Return(array);
             }
         }
+
+        public static async Task CopyToAsync(this Stream source, Stream destination, CancellationToken cancellationToken)
+        {
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(DefaultCopyBufferSize);
+
+            try
+            {
+                while (true)
+                {
+#pragma warning disable CA1835 // ReadAsync(Memory<>) overload is not available in all targets
+                    int bytesRead = await source.ReadAsync(buffer, 0, buffer.Length, cancellationToken).ConfigureAwait(false);
+#pragma warning restore // ReadAsync(Memory<>) overload is not available in all targets
+                    if (bytesRead == 0)
+                        break;
+                    await destination.WriteAsync(new ReadOnlyMemory<byte>(buffer, 0, bytesRead), cancellationToken).ConfigureAwait(false);
+                }
+
+                await destination.FlushAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Added this in response to https://github.com/Azure/azure-sdk-for-net/pull/41773/files#r1491336131

Reply:
does it make sense for this to go first in the finally block?

Reply:
Makes sense - good idea